gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Client.java

```diff
@@ -750,7 +754,9 @@ protected void initializeImplementation() {
             final Host host = hosts.get(0);
 
             try {
+                noLiveHostAvailable = true;
```

Comment:
Should this be in the catch?

Reply:
I've changed the logic so that we are setting noLiveHostAvailable to be true at initiation, and change it to false only when we succeed on at least one connection in cluster clients. I don't think this matter for session clients (which is the one here) whether this goes into the try or the catch, but I've put it up here for consistency. Unless I'm missing a logic that could make a difference?

Reply:
I think this is correct. Logic is inverted from typical try catch flag setting.
This is doing, assume it won't work, then if it does change it to say it did work.
Makes sense in this case.